crates/aggkit-prover/src/aggchain_proof/service.rs

```diff
@@ -4,71 +4,116 @@ use std::{
+use proposer_service::{ProposerRequest, ProposerService};
+use tracing::debug;
 
 use super::error::Error;
 
-pub struct ProofRequest {
-    #[allow(dead_code)]
+/// Request to the AggchainProofService to generate the
+/// aggchain proof for the range of blocks.
+#[derive(Default, Clone, Debug)]
+#[allow(unused)]
+pub struct AggchainProofServiceRequest {
+    /// Aggchain proof starting block
     pub start_block: u64,
-    #[allow(dead_code)]
+    /// Max number of blocks that the aggchain proof could contain
     pub max_block: u64,
+    /// L1 Info tree root hash.
+    pub l1_info_tree_root_hash: Hash,
+    /// L1 Info tree leaf hash.
+    pub l1_info_tree_leaf_hash: Hash,
+    /// L1 Info tree proof.
+    pub l1_info_tree_merkle_proof: [Hash; 32],
 }
 
-pub struct ProofResponse {}
+/// Resulting generated Aggchain proof
+pub struct AggchainProofServiceResponse {
+    pub proof: AggchainProof,
+    pub start_block: u64,
+    pub end_block: u64,
+}
 
 /// The Aggchain proof service is responsible for orchestrating an Aggchain
 /// proof generation.
 ///
 /// The Aggchain proof is generated by fetching the Aggregated FEP from the
-/// proposer service and the aggchain proof builder service to generate the
+/// proposer service and the `aggchain-proof-builder` service to generate the
 /// Aggchain proof.
 #[derive(Clone)]
 pub(crate) struct AggchainProofService {
     pub(crate) proposer_service: ProposerService,
-    pub(crate) aggchain_proof_builder: AggchainProofBuilderService,
+    pub(crate) aggchain_proof_builder: AggchainProofBuilder,
 }
 
-impl Default for AggchainProofService {
-    fn default() -> Self {
-        // Implement new function and remove this default
-        todo!()
+impl AggchainProofService {
+    pub fn new(config: &AggchainProofServiceConfig) -> Result<Self, Error> {
+        Ok(AggchainProofService {
+            proposer_service: ProposerService::new(&config.proposer_service_config)?,
+            aggchain_proof_builder: AggchainProofBuilder::new(
+                &config.aggchain_proof_builder_config,
+            )?,
+        })
     }
 }
 
-impl tower::Service<ProofRequest> for AggchainProofService {
-    type Response = ProofResponse;
+impl tower::Service<AggchainProofServiceRequest> for AggchainProofService {
+    type Response = AggchainProofServiceResponse;
     type Error = Error;
     type Future = Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send>>;
 
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        if self
-            .proposer_service
-            .poll_ready(cx)
-            .map_err(|_| Error::Noop)
-            .is_pending()
-        {
-            return Poll::Pending;
-        }
+        match self.proposer_service.poll_ready(cx) {
+            Poll::Ready(Ok(())) => (),
+            Poll::Ready(Err(e)) => return Poll::Ready(Err(e.into())),
+            Poll::Pending => return Poll::Pending,
+        };
```

Comment:
```suggestion
        std::task::ready!(self.proposer_service.poll_ready(cx)?);
```
See explanation here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1dc4200a350cc1e711672625872fa05a

Reply:
Nice!